viser/_message_api.py

```diff
@@ -70,6 +71,45 @@ def _colors_to_uint8(colors: onp.ndarray) -> onpt.NDArray[onp.uint8]:
 ]
 
 
+class TitlebarButton(dict):
+    def __init__(
+        self,
+        text: Optional[str] = None,
+        icon: Optional[str] = None,
```

Comment:
for `icon` and `variant` can we annotate with a `typing.Literal[]`? that specifies the allowable strings